tests/backend/core/Core/Unit/Logic/Segment/RpcBulkEditor/SegmentBulkEditorServiceTest.php

```diff
@@ -0,0 +1,161 @@
+use demosplan\DemosPlanCoreBundle\Logic\Segment\SegmentBulkEditorService;
+use Doctrine\ORM\EntityManagerInterface;
+use Tests\Base\RpcApiTest;
+
+class SegmentBulkEditorServiceTest extends RpcApiTest
+{
+    /** @var SegmentBulkEditorService */
+    protected $sut;
+    private $procedure;
+    private $segment1;
+    private $segment2;
+    private $entityManager;
+    private $entityType;
+
+    private $user;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->sut = $this->getContainer()->get(SegmentBulkEditorService::class);
+        $this->procedure = $this->getProcedureReference(LoadProcedureData::TESTPROCEDURE);
+        $this->segment1 = $this->getSegmentReference(LoadSegmentData::SEGMENT_BULK_EDIT_1);
+        $this->segment2 = $this->getSegmentReference(LoadSegmentData::SEGMENT_BULK_EDIT_2);
+        $this->entityManager = $this->getContainer()->get(EntityManagerInterface::class);
+        $this->entityType = $this->entityManager->getClassMetadata(Segment::class)->getName();
+        $this->user = $this->loginTestUser();
+        $this->prepareSegments();
+    }
+
+    private function prepareSegments(): void
+    {
+        $em = $this->getEntityManager();
+        $this->segment1->setRecommendation('Initial text 1');
+        $this->segment2->setRecommendation('Initial text 2');
+        $em->persist($this->segment1);
+        $em->persist($this->segment2);
+        $em->flush();
+    }
+
+    public function testUpdateSegmentsWithNotNullAssignee(): void
+    {
+        self::assertNull($this->segment1->getAssignee());
+        self::assertNull($this->segment2->getAssignee());
+
+        $this->sut->updateSegments([$this->segment1, $this->segment2], [], [], $this->user, null);
+
+        self::assertEquals($this->user->getId(), $this->segment1->getAssignee()->getId());
+        self::assertEquals($this->user->getId(), $this->segment2->getAssignee()->getId());
+    }
+
+    public function testUpdateSegmentsWithNullAssignee(): void
+    {
+        $this->segment1->setAssignee($this->user);
+        $this->segment2->setAssignee($this->user);
+
+        self::assertEquals($this->user->getId(), $this->segment1->getAssignee()->getId());
+        self::assertEquals($this->user->getId(), $this->segment2->getAssignee()->getId());
+
+        $this->sut->updateSegments([$this->segment1, $this->segment2], [], [], null, null);
+
+        self::assertNull($this->segment1->getAssignee());
+        self::assertNull($this->segment2->getAssignee());
+    }
+
+    public function testDetectValidAssignee(): void
+    {
+        $user = $this->loginTestUser();
+        $assignee = $this->sut->detectAssignee($user->getId());
+        self::assertEquals($user->getId(), $assignee->getId());
+    }
+
+    public function testDetectInvalidIdAssignee(): void
+    {
+        self::expectException(UserNotFoundException::class);
+        $invalidAssigneeId = '123';
+        $this->sut->detectAssignee($invalidAssigneeId);
```

Comment:
We don't need any assert like assertNull here?

Reply:
done bb2fefc7b61d51f164aeac8f977a779a7f312917